Significant-Gravitas/Auto-GPT-Benchmarks
# fixing memory challenges, naming, testing mini-agi, smooth retrieval scaling

### Background
- Fixing bug with memory challenges where we were copying the artifacts into the workspace and then checking all of the files if the id exists - and it does within the files we copied into the workspace at the end of the run
- Naming different challenges
- Testing mini-agi with many different challenges
- Making retrieval scaling more smooth (added 2.0, 2.1, 2.2)

TODO 
1. Figure out why mini-agi says the Python file has no errors but it still fails. Also the same bug should be present here as the memory challenge but I'm not sure why it isn't. 
2. Make retrieval challenges one challenge that has a success scale from 0, 33%, 66%, 100% instead of 2.0, 2.1, 2.2/
3. Add UI of what challenges do or don't pass to the README.md

```
MiniAGI: The latest execution of test.py returned no errors, which is aligned with our goal. Therefore, we can conclude that the objective has been achieved.
Cmd: done, Arg:

The Python function has finished running.
Scoring content:  
Word that should exist - [0, 1]: False
Your score is: 0.0
```

### PR Quality Checklist
- [x] I have run the following commands against my code to ensure it passes our linters:
    ```shell
    black .
    isort .
    mypy .
    autoflake --remove-all-unused-imports --recursive --ignore-init-module-imports --ignore-pass-after-docstring --in-place agbenchmark
    ```
